src/sdk/namespace_blob.js

```diff
@@ -395,6 +395,16 @@ class NamespaceBlob {
     }
 
     _check_valid_xattr(params) {
+
+        const xattr = params.xattr;
+        for (const xattrKey in xattr) {
+            const tmp = xattrKey.replace(/-/g, '_');
+            const tmp2 = xattr[xattrKey];
+            delete xattr[xattrKey];
+            xattr[tmp] = tmp2;
+        }
+        params.xattr = xattr;
```

Comment:
Will this work instead? 
```suggestion
        for (const xattrKey in params.xattr) {
            params.xattr[xattrKey] = params.xattr[xattrKey].replace(/-/g, '_');
        }
```

Reply:
maybe even:
```suggestion
        for (const [key, val] of params.xattr) {
            params.xattr[key] = val.replace(/-/g, '_');
        }
```